components/payments/internal/messages/transfer_initiations.go

```diff
@@ -51,9 +50,13 @@ func (m *Messages) NewEventSavedTransferInitiations(tf *models.TransferInitiatio
 		DestinationAccountID: tf.DestinationAccountID.String(),
 		Amount:               tf.Amount,
 		Asset:                tf.Asset,
-		Attempts:             tf.Attempts,
-		Status:               tf.Status.String(),
-		Error:                tf.Error,
+		Attempts:             len(tf.RelatedAdjustments),
+	}
+
+	if len(tf.RelatedAdjustments) > 0 {
```

Comment:
Maybe, you could have created a custom MarshalJSON() function on the response objet to automatically set those fields.